app/models/measurement_list_reader.rb

```diff
@@ -23,15 +23,17 @@ def load
 
   private
 
-  def load_measurements_from_gr
+  def load_measurements_from_gr # rubocop:disable Metrics/MethodLength
     work_q = Queue.new
     params = gr_loading_params
     @measurements.each { |m| work_q.push m }
     workers = (1..thread_count).map do
       Thread.new do
         begin
           loop do
-            Measurement::GrValueLoader.new(params.merge(measurement: work_q.pop(true))).load_gr_value
+            Rails.application.reloader.wrap do
+              Measurement::GrValueLoader.new(params.merge(measurement: work_q.pop(true))).load_gr_value
+            end
```

Comment:
I had to wrap this line in this block otherwise I would get Circular Dependency errors.